freecad/cross/utils.py

```diff
@@ -186,35 +190,33 @@ def true_then_false(list: Iterable[bool]) -> bool:
     False
 
     """
-    if not list:
-        return True
-    v0 = list[0]
-    for v in list:
-        if v0 or not v:
-            v0 = v
-            continue
-        return False
-    return True
+    return reduce(lambda a,b: (a[0] and a[1] >= b, b), list, (True, True))[0]
```

Comment:
builtins are more concise and performant.

Reply:
This one is more complex but also very nice. Didn't know or forgot about `reduce`.  Never did a comparison between two booleans, it may be useful in some situations but looks a little bit like a hack.

If you happen to make some changes, please add a space: `a, b`. I also notice that `list` should be renamed `list_`.